src/utils/LoggingInterface.ts

```diff
@@ -0,0 +1,60 @@
+
+/**
+ * Logging interface for the model generation library
+ */
+export interface ModelLoggingInterface {
+  debug(message?: any, ...optionalParams: any[]): void;
+  info(message?: any, ...optionalParams: any[]): void;
+  warn(message?: any, ...optionalParams: any[]): void;
+  error(message?: any, ...optionalParams: any[]): void;
+}
+
+/**
+ * Logger class for the model generation library
+ * 
+ * This class acts as a forefront for any external loggers which is why it also implements the interface itself.
+ */
+export class Logger implements ModelLoggingInterface {
+  private static instance?: Logger;
```

Comment:
I don't think so this is a good idea. We should create default logger by in constructor, and avoid logic like `Logger.getInstance().*`. As developer I rather prefer `Logger.*`

Reply:
Hmm, good point, let me refactor it.